src/test/java/com/iota/iri/crypto/ISSTest.java

```diff
@@ -78,6 +78,28 @@ public void addressGenerationISS() throws Exception {
         }
     }
 
+    @Test
+    public void sigDigestISSInPlace() throws Exception {
+        int index = 0;
+        int nof = 1;
+        SpongeFactory.Mode[] modes = {SpongeFactory.Mode.KERL};
+        Hash[] hashes = {HashFactory.ADDRESS.create("DQMAVYJTIIQUCDHRIZAPYKFCFBTHNDAMHLYEEIEBSR9REQBBVUQOSDWFSGPTZFNQQTVMQPVUDHDTEZVSW")};
+        for (int i=0; i < modes.length; i++) {
+            SpongeFactory.Mode mode = modes[i];
+            byte[] seedTrits = Converter.allocateTritsForTrytes(seed.length());
+            Converter.trits(seed, seedTrits, 0);
+
+            byte[] subseed = ISS.subseed(mode, seedTrits, index);
+            byte[] key = ISS.key(mode, subseed, nof);
+            byte[] digest = new byte[Curl.HASH_LENGTH];
+            byte[] sig = Arrays.copyOfRange(key, 0, key.length);
+            ISSInPlace.digest(mode, sig, 0, key, 0, digest);
+            byte[] address = ISS.address(mode, digest);
+            Hash addressTrytes = HashFactory.ADDRESS.create(address);
+            assertEquals(hashes[i].toString(), addressTrytes.toString());
```

Comment:
1. Can you please add a message in the first parameter? i.e.  "Address is not as expected"
2. The `toString()` is extraneous, `equals()` is defined :-)

Reply:
which function's first parameter? 
will remove toString(), again, the existing function also used toString(), should be removed for consistency.

Reply:
@ed2k every `assert()` function accepts a first parameter with a message to display when the test fails. Thats the one @GalRogozinski meant.